src/hammer-vlsi/cli_driver.py

```diff
@@ -121,6 +129,23 @@ def create_par_action(self, custom_hooks: List[HammerToolHookAction],
         return self.create_action("par", hooks if len(hooks) > 0 else None,
                                   pre_action_func, post_load_func, post_run_func)
 
+    def create_formal_action(self, custom_hooks: List[HammerToolHookAction],
+                                pre_action_func: Optional[Callable[[HammerDriver], None]] = None,
+                                post_load_func: Optional[Callable[[HammerDriver], None]] = None,
+                                post_run_func: Optional[Callable[[HammerDriver], None]] = None
+                                ) -> CLIActionType:
+        hooks = self.get_extra_synthesis_hooks() + custom_hooks  # type: List[HammerToolHookAction]
+        return self.create_action("formal", hooks if len(hooks) > 0 else None,
+                                  pre_action_func, post_load_func, post_run_func)
+
+    def create_power_action(self, custom_hooks: List[HammerToolHookAction],
+                                pre_action_func: Optional[Callable[[HammerDriver], None]] = None,
+                                post_load_func: Optional[Callable[[HammerDriver], None]] = None,
+                                post_run_func: Optional[Callable[[HammerDriver], None]] = None
+                                ) -> CLIActionType:
+        hooks = self.get_extra_synthesis_hooks() + custom_hooks  # type: List[HammerToolHookAction]
```

Comment:
These extra hooks should use `get_extra_power_hooks` method or the `get_extra_formal_hooks` right?